### PR TITLE
Add text matching utility function

### PR DESCRIPTION
**Pull request by nmanu1**
Adds the `isCloseMatch` function in answers-utilities to perform case-insensitive Levenshtein matching. This is used in `searchThroughFacet`.

J=SLAP-1802
TEST=auto

See that jest tests for `searchThroughFacet` pass.

### CODE REVIEW

File: src/answers-utilities.ts [hunk 1]
```diff
@@ -14,9 +14,20 @@ export default {
...
-      options: facet.options.filter(o =>
-        isLevenshteinMatch(o.displayName.toLowerCase(), searchTerm.toLowerCase())
-      )
+      options: facet.options.filter(o => isCloseMatch(o.displayName, searchTerm))
     };
-  }
-};
\ No newline at end of file
+  },
+  isCloseMatch
```
Comment (tmeyer2115) [inline src/answers-utilities.ts:20]: Does this export work properly? There can only be one `default` export per module. It looks as if we're attempting to have two default exports. If this module is to export both `searchThroughFacet` and `isCloseMatch`, one should either be the `default` or we should have two named exports.

Comment (oshi97) [reply src/answers-utilities.ts:20]: this is exporting an single object that has two functions. I think I would just prefer 2 named exports, and an `import * as AnswersUtilities from './answers-utilities` wherever this gets imported if we really want to group things

Comment (nmanu1) [reply src/answers-utilities.ts:20]: It's not giving any errors with the export like this and I'm able to use it in the starter repo locally. But, I can switch to having two named exports instead. That would also remove the forgotten exports warning in the API report.

**Comment by tmeyer2115** [comment]
Remember to update the target branch here. We can no longer merge directly into `main`.

**Review by oshi97 — APPROVED** [review]

**Pull request merged**